### ISSUE TITLE
Question: Empty query output

### ISSUE THREAD

**Issue by langecode** [jacksontj/promxy#285]
This is most probably just for the lack of understanding the configuration. I am experimenting with a setup containing 3 server-groups as I intend to do a gradual migration. Basically one server-group will contain two VictoriaMetrics instances forming a HA setup, the other two groups each contains an existing VictoriaMetrics instance.

Doing a very simple range query data will only be found in one server-group but I get back an empty result even if the log shows data was actually found. Promxy version i `0.0.58`.

Configuration:
```yaml
    global:
      evaluation_interval: 5s
      external_labels:
        source: promxy
    promxy:
      server_groups:
      - static_configs:
        - targets:
          - vm-tm-0-victoria-metrics-single-server:8428
          - vm-tm-1-victoria-metrics-single-server:8428
        labels:
          sg: vm_single
      - static_configs:
        - targets:
          - victoria-metrics-vmselect:8481
        path_prefix: /select/1/prometheus
        labels:
          sg: vm_cluster_1
      - static_configs:
        - targets:
          - victoria-metrics-vmselect:8481
        path_prefix: /select/2/prometheus
        labels:
          sg: vm_cluster_2
```

Query: `http://localhost:8082/api/v1/query_range?query=Model1NRE_BearTemp1_value%7BTurbineId%3D%22210016%22%7D&start=1584364290.979&end=1584450690.979&step=345&_=1584439227432`

Response:
```json
{
  "status":"success",
  "data": {
    "resultType":"matrix",
    "result":[]
  }
}
```

Log output (masked the actual data a little bit):
```
time="2020-03-17T13:11:31Z" level=debug msg="VectorSelector: Model1NRE_BearTemp1_value{TurbineId=\"210016\"}"
time="2020-03-17T13:11:31Z" level=debug msg="http://victoria-metrics-vmselect:8481/select/2/prometheus" api=QueryRange query="Model1NRE_BearTemp1_value{TurbineId=\"210016\"}" r="{2020-03-16 13:11:30.979 +0000 UTC 2020-03-17 13:11:30.979 +0000 UTC 5m45s}"
time="2020-03-17T13:11:31Z" level=debug msg="http://victoria-metrics-vmselect:8481/select/1/prometheus" api=QueryRange query="Model1NRE_BearTemp1_value{TurbineId=\"210016\"}" r="{2020-03-16 13:11:30.979 +0000 UTC 2020-03-17 13:11:30.979 +0000 UTC 5m45s}"
time="2020-03-17T13:11:31Z" level=debug msg="http://vm-tm-1-victoria-metrics-single-server:8428" api=QueryRange query="Model1NRE_BearTemp1_value{TurbineId=\"210016\"}" r="{2020-03-16 13:11:30.979 +0000 UTC 2020-03-17 13:11:30.979 +0000 UTC 5m45s}"
time="2020-03-17T13:11:31Z" level=debug msg="http://vm-tm-0-victoria-metrics-single-server:8428" api=QueryRange query="Model1NRE_BearTemp1_value{TurbineId=\"210016\"}" r="{2020-03-16 13:11:30.979 +0000 UTC 2020-03-17 13:11:30.979 +0000 UTC 5m45s}"
time="2020-03-17T13:11:31Z" level=trace msg="http://vm-tm-0-victoria-metrics-single-server:8428" api=QueryRange error="<nil>" query="Model1NRE_BearTemp1_value{TurbineId=\"210016\"}" r="{2020-03-16 13:11:30.979 +0000 UTC 2020-03-17 13:11:30.979 +0000 UTC 5m45s}" took=1.883807ms value= warnings="[]"
time="2020-03-17T13:11:31Z" level=trace msg="http://vm-tm-1-victoria-metrics-single-server:8428" api=QueryRange error="<nil>" query="Model1NRE_BearTemp1_value{TurbineId=\"210016\"}" r="{2020-03-16 13:11:30.979 +0000 UTC 2020-03-17 13:11:30.979 +0000 UTC 5m45s}" took=1.982208ms value= warnings="[]"
time="2020-03-17T13:11:31Z" level=trace msg="http://victoria-metrics-vmselect:8481/select/1/prometheus" api=QueryRange error="<nil>" query="Model1NRE_BearTemp1_value{TurbineId=\"210016\"}" r="{2020-03-16 13:11:30.979 +0000 UTC 2020-03-17 13:11:30.979 +0000 UTC 5m45s}" took=2.242208ms value= warnings="[]"
time="2020-03-17T13:11:31Z" level=trace msg="http://victoria-metrics-vmselect:8481/select/2/prometheus" api=QueryRange error="<nil>" query="Model1NRE_BearTemp1_value{TurbineId=\"210016\"}" r="{2020-03-16 13:11:30.979 +0000 UTC 2020-03-17 13:11:30.979 +0000 UTC 5m45s}" took=5.139318ms value="Model1NRE_BearTemp1_value{ClusterId=\"**\", ClusterName=\"G**\", ModelId=\"**\", ModelName=\"A1**_BearTemp1\", Parameter=\"Average\", ParkId=\"**\", ParkName=\"**\", TurbineId=\"**\", TurbineName=\"**\", TurbineType=\"V**\", db=\"ModelResult\", sg=\"vm_cluster_2\"} =>\n43.4 @[1584374550]" warnings="[]"
```

**Comment by jacksontj** [comment]
This is almost certainly the internal caching in VM -- you can disable it at query time (example config -- https://github.com/jacksontj/promxy/blob/master/cmd/promxy/config.yaml#L53) which should solve the issue for you :)

**Comment by langecode** [comment]
Thanks a lot and thanks for the answer, it _does_ solve the issue :-)

I am struggling a bit to understand why that is when the log show that data was returned from one of the VM instances? We did already disable caching on the VM side since we are ingesting data from IoT devices which may at times be offline, often resulting in back-filling of data.

**Comment by jacksontj** [comment]
I'm glad that it solved your issue, it is the #1 issue with promxy+VM :)

Well, from the promxy side the issue isn't that it was returning "no data"
but that it was returning the "wrong data" specifically that the timestamps
weren't what it was requesting. VM internally (similar to trickster)
rewrites the query timestamps to improve cache hit rates -- but it does
mean that the result you get isn't exactly what you asked for. For some
clients (e.g. grafana) the time being off is okay but for others that do
calculations (e.g. promxy) this means the data is unusable.

As for why disabling caching in VM and adding the nocache query param
behave differently my guess would be that disabling caching doesn't disable
the rewriting of times -- which seems like non-obvious behavior, but for
that you'd need to create an issue over with VM so they can decide if thats
intended or not :)

**Comment by langecode** [comment]
Makes, sense. Thanks for your help :)
